Description: Fix the disappearance of item names during the game
Author: Fabian Wolff <fabi.wolff@arcor.de>
Forwarded: no
Last-Update: 2016-06-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/brogue/Items.c
+++ b/src/brogue/Items.c
@@ -1325,13 +1325,11 @@
 				if (theItem->flags & ITEM_RUNIC) {
 					if ((theItem->flags & ITEM_RUNIC_IDENTIFIED) || rogue.playbackOmniscience) {
 						if (theItem->enchant2 == W_SLAYING) {
-							sprintf(root, "%s of %s slaying%s",
-									root,
-                                    monsterClassCatalog[theItem->vorpalEnemy].name,
+							sprintf (root + strlen (root), " of %s slaying%s",
+									monsterClassCatalog[theItem->vorpalEnemy].name,
 									grayEscapeSequence);
 						} else {
-							sprintf(root, "%s of %s%s",
-									root,
+							sprintf(root + strlen (root), " of %s%s",
 									weaponRunicNames[theItem->enchant2],
 									grayEscapeSequence);
 						}
@@ -1342,7 +1340,7 @@
 						strcat(root, " (unknown runic)");
 					}
 				}
-				sprintf(root, "%s%s <%i>", root, grayEscapeSequence, theItem->strengthRequired);
+				sprintf(root + strlen (root), "%s <%i>", grayEscapeSequence, theItem->strengthRequired);
 			}
 			break;
 		case ARMOR:
@@ -1353,9 +1351,11 @@
 					&& ((theItem->flags & ITEM_RUNIC_IDENTIFIED)
 						|| rogue.playbackOmniscience)) {
 						if (theItem->enchant2 == A_IMMUNITY) {
-							sprintf(root, "%s of %s immunity", root, monsterClassCatalog[theItem->vorpalEnemy].name);
+							sprintf(root + strlen (root), " of %s immunity",
+								monsterClassCatalog[theItem->vorpalEnemy].name);
 						} else {
-							sprintf(root, "%s of %s", root, armorRunicNames[theItem->enchant2]);
+							sprintf(root + strlen (root), " of %s",
+								armorRunicNames[theItem->enchant2]);
 						}
 					}
 				
@@ -1373,7 +1373,9 @@
 					}
 					strcpy(root, buf);
 				} else {
-					sprintf(root, "%s%s <%i>", root, grayEscapeSequence, theItem->strengthRequired);
+					sprintf(root + strlen (root), "%s <%i>",
+						grayEscapeSequence,
+						theItem->strengthRequired);
 				}
 				
 				if ((theItem->flags & ITEM_RUNIC)
@@ -1438,18 +1440,15 @@
 			}
 			if (includeDetails) {
 				if (theItem->flags & (ITEM_IDENTIFIED | ITEM_MAX_CHARGES_KNOWN) || rogue.playbackOmniscience) {
-					sprintf(root, "%s%s [%i]",
-							root,
+					sprintf(root + strlen (root), "%s [%i]",
 							grayEscapeSequence,
 							theItem->charges);
 				} else if (theItem->enchant2 > 2) {
-					sprintf(root, "%s%s (used %i times)",
-							root,
+					sprintf(root + strlen (root), "%s (used %i times)",
 							grayEscapeSequence,
 							theItem->enchant2);
 				} else if (theItem->enchant2) {
-					sprintf(root, "%s%s (used %s)",
-							root,
+					sprintf(root + strlen (root), "%s (used %s)",
 							grayEscapeSequence,
 							(theItem->enchant2 == 2 ? "twice" : "once"));
 				}
@@ -1473,9 +1472,9 @@
 			}
 			if (includeDetails) {
 				if ((theItem->flags & ITEM_IDENTIFIED) || rogue.playbackOmniscience) {
-					sprintf(root, "%s%s [%i/%i]", root, grayEscapeSequence, theItem->charges, theItem->enchant1);
+					sprintf(root + strlen (root), "%s [%i/%i]", grayEscapeSequence, theItem->charges, theItem->enchant1);
 				} else if (theItem->flags & ITEM_MAX_CHARGES_KNOWN) {
-					sprintf(root, "%s%s [?/%i]", root, grayEscapeSequence, theItem->enchant1);
+					sprintf(root + strlen (root), "%s [?/%i]", grayEscapeSequence, theItem->enchant1);
 				}
 			}
 			break;
